src/context/WalletProvider/NewWalletViews/routes/MipdRoutes.tsx

```diff
@@ -0,0 +1,53 @@
+import { Route, Switch } from 'react-router-dom'
+import { useWallet } from 'hooks/useWallet/useWallet'
+
+import { SnapInstall } from '../../MetaMask/components/SnapInstall'
+import { SnapUpdate } from '../../MetaMask/components/SnapUpdate'
+import { RDNS_TO_FIRST_CLASS_KEYMANAGER } from '../constants'
+import type { RightPanelContentProps } from '../types'
+import { FirstClassBody } from '../wallets/mipd/FirstClassBody'
+import { MipdBody } from '../wallets/mipd/MipdBody'
+
+export const MipdRoutes = ({
+  isLoading,
+  error,
+  setIsLoading,
+  setError,
+}: Omit<RightPanelContentProps, 'location'>) => {
+  const {
+    state: { modalType },
+  } = useWallet()
+
+  if (!modalType) return null
+
+  return (
+    <Switch>
+      {Object.values(RDNS_TO_FIRST_CLASS_KEYMANAGER).map(keyManager => (
+        <Route exact path={`/${keyManager.toLowerCase()}/connect`}>
+          <FirstClassBody
+            keyManager={keyManager}
+            isLoading={isLoading}
+            error={error}
+            setIsLoading={setIsLoading}
+            setError={setError}
+          />
+        </Route>
+      ))}
+      <Route exact path='/metamask/connect'>
+        <MipdBody
+          rdns={modalType}
+          isLoading={isLoading}
+          error={error}
+          setIsLoading={setIsLoading}
+          setError={setError}
+        />
+      </Route>
+      <Route path='/metamask/snap/install'>
+        <SnapInstall />
+      </Route>
+      <Route path='/metamask/snap/update'>
```

Comment:
Actually, there *is* another change in addition to the first-class handling `/metamask/failure` handling is now unhandled, as we don't push to it anymore in `<FirstClassBody />` / `<MipdBody />`.

The rationale is there is no need for an explicit pairing failure step, errors are handled within the body itself 

https://jam.dev/c/f6c73f5b-d68e-491a-a741-d7463cfe2870